### ISSUE TITLE
WSL2 - DNS error on telepresence connect

### ISSUE THREAD

**Issue by lizfrench-esgglobal** [telepresenceio/telepresence#3209]
**Describe the bug**
When I call 'telepresence connect' i get the following feedback:

Connected to context rancher-desktop (https://<ipaddress>:6443)
**Warning: DNS doesn't seem to work properly**

It does appear to connect but I cannot do any nslookups.

**To Reproduce**
I have the following installed:

1) Rancher Desktop - Version 1.8.1 / Configured in the settings to run kubernetes and linked to WSL2.
2) WSL2 - version 22.04
3) telepresence linux distribution via the following command:
sudo curl -fL https://app.getambassador.io/download/tel2/linux/amd64/latest/telepresence -o /usr/local/bin/telepresence

1. When I run 'telepresence connect'
2. I see the response of 'Connected to cont
ext rancher-desktop (https://<ipaddress>:6443)
**Warning: DNS doesn't seem to work properly**'
3. So I try nslookup <svc.namespace>
4. And nothing is found

**Expected behavior**
The DNS warning should not appear and the nslookup should resolve.

**Versions (please complete the following information):**
 - Output of `telepresence version` => 2.13.3
 - Operating system of workstation running `telepresence` commands => Linux / WSL2 V22.04
 - Kubernetes environment and Version [Rancher Desktop V1.8.1 which runs - KS3]

**Additional context**
Attended Tuesday evening session with Cindy and Jay - thank you for looking at this, it's really important for us.  If I can get this running it will speed up our development loop and give me a good case to recommend telepresence to be adopted by our company.

**Logfile**

[telepresence_logs.zip](https://github.com/telepresenceio/telepresence/files/11604571/telepresence_logs.zip)

**Comment by koshkarov** [comment]
I'm experiencing this too. I'll attach the logs shortly!

**Comment by cindymullins-dw** [comment]
On Windows/WSL could you try running `Ipconfig /flushdns` and then `Ipconfig /displaydns`, and let us know the output?  We’re trying to see if DNS is populating the IP tables correctly.  It’d be interesting also to see if running that before and then after `Telepresence connect` makes any difference.

**Comment by koshkarov** [comment]
@cindymullins-dw 

I have the same issue but with WSL2 (Ubuntu 22.04). 

Steps that I tried to resolve the problem:
- Reinstalled the Ubuntu 22.04 distro, to exclude all installation-specific issues.
- Installed the latest version of the traffic manager (v2.13.3). 
- Installed the latest version of the telepresence (v2.13.3). 
The cluster is running in GKE and access protected with [Teleport](https://goteleport.com/).

In PowerShell ran commands `Ipconfig /flushdns` and `Ipconfig /displaydns`. The output is attached in **dns.txt**.

In WSL2 (Ubuntu 22.04):
 - Switched to the correct context;
 - Ran `telepresence connect` and received the output:
```
Launching Telepresence User Daemon
Launching Telepresence Root Daemon
Need root privileges to run: /usr/local/bin/telepresence daemon-foreground /home/koshkarov/.cache/telepresence/logs /home/koshkarov/.config/telepresence
[sudo] password for koshkarov:
Connected to context <hidden>-development-cluster-1 (https://<hidden>.com:3026)
Warning: DNS doesn't seem to work properly
```

The logs in **telepresence-logs.zip**.

Asking for help as I'm totally frustrated by the issue. It looks like the problem in Windows DNS configuration, but how to debug the problem I have no idea.

[dns.txt](https://github.com/telepresenceio/telepresence/files/11638131/dns.txt)
[telepresence-logs.zip](https://github.com/telepresenceio/telepresence/files/11638129/telepresence-logs.zip)

Then after running ` telepresence intercept vlad-event-service -n vlad-event-service --port=5000:http --env-file='.env'` with output:
```
Using Deployment vlad-event-service
   Intercept name         : vlad-event-service-vlad-event-service
   State                  : ACTIVE
   Workload kind          : Deployment
   Destination            : 127.0.0.1:5000
   Service Port Identifier: http
   Volume Mount Error     : sshfs is not installed on your local machine
   Intercepting           : all TCP requests
Intercepting all traffic to your service. To route a subset of the traffic instead, use a personal intercept. You can enable personal intercepts by authenticating to Ambassador Cloud with "telepresence login".
```

And to check if that worked: `curl -ik https://kubernetes.default` returned:
```
curl: (6) Could not resolve host: kubernetes.default
```

**Comment by reda-semlal-tas** [comment]
Are there any updates on this issue?

**Comment by declanoshanahan** [comment]
For anybody else experiencing this issue. I am running **rancher-desktop 1.9.1, telepresence version 2.14.1 and Ubuntu 22.04.2**  LTS. I was able to work around DSN issues when connecting to a rancher-desktop 1.9.1 installation when using telepresence 2.14.1 on Ubuntu 22.04.2 where the ubuntu installation was **using the DNS resolver systemd-resolved**

**lsb_release -a**
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.2 LTS
Release:	**22.04**
Codename:	jammy

**telepresence version**
Client         : **v2.14.1**
Root Daemon    : **v2.14.1**
User Daemon    : **v2.14.1**
Traffic Manager: **v2.14.1**

I deleted and recreated the soft link the /etc/resolv.conf file to point to the  /run/systemd/resolve/resolv.conf instead of the  /run/systemd/resolve/stub-resolv.conf and issued a restart of the systemd-resolved service as follows

sudo rm -f /etc/resolv.conf
sudo ln -s /run/systemd/resolve/resolv.conf /etc/resolv.conf
sudo service systemd-resolved restart
ls -ltr /etc/resolv.conf 
lrwxrwxrwx 1 root root 32 Aug  1 14:05 /etc/resolv.conf -> /run/systemd/resolve/resolv.conf

I then issued the following commands to quit and reconnect to the rancher-desktop environment using telepresence. For the resolvectl output I have replaced the names of our internal DNS Domain with a dummy DNS name. My company is running an VPN internally and so I have specify the subnets for the rancher-desktop services and api subnet ranges to work around the VPN routing rules already in place. 

telepresence helm uninstall
telepresence helm install --set client.routing.allowConflictingSubnets='{10.43.0.0/16,10.42.0.0/16}'
telepresence quit -s
telepresence connect

**resolvectl status | grep -A 10 tun0**
Link 10 (zcctun0)
    Current Scopes: DNS
         Protocols: +DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported
Current DNS Server: 100.64.0.2
       DNS Servers: 100.64.0.2
        DNS Domain: dummy.com dummy.net dummy.net ~.

Link 12 (tel0)
Current Scopes: none
     Protocols: -DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported

Querying the namespace core-example-ft-lab I can get the service name int-vip-core-example 

kubectl -n core-example-ft-lab get services -o wide
NAME                   TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)             AGE   SELECTOR
int-vip-core-example   ClusterIP   10.43.234.120   <none>        6060/TCP,6061/TCP   25h   app.kubernetes.io/instance=core-example-ft-lab,app.kubernetes.io/name=core-example

At this point I can then resolve the internal service name e.g. int-vip-core-example.core-example-ft-lab as follows.

nslookup int-vip-core-example.core-example-ft-lab
Server:		100.64.0.2
Address:	100.64.0.2#53

Name:	int-vip-core-example.core-example-ft-lab
Address: 10.43.234.120

I can then also now use the service name of the form to issue a curl command to query the internal service int-vip-core-example.core-example-ft-lab running a management springboot endpoint on port 6061 as follows.

curl http://int-vip-core-example.core-example-ft-lab:6061/management/info
*   Trying 10.43.234.120:6061...
* Connected to int-vip-core-example.core-example-ft-lab (10.43.234.120) port 6061 (#0)
> GET /management/info HTTP/1.1
> Host: int-vip-core-example.core-example-ft-lab:6061
> User-Agent: curl/7.81.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 
< Content-Type: application/vnd.spring-boot.actuator.v3+json
< Transfer-Encoding: chunked
< Date: Tue, 01 Aug 2023 14:45:05 GMT
< 
* Connection #0 to host int-vip-core-example.core-example-ft-lab left intact
{"build":{"version":"1.0.0-DEV","artifact":"core-example","name":"core-example","group":"com.dummy.example","time":"2023-07-31T12:29:45.660Z"}}

**Comment by shinobu-uwu** [comment]
Any updates?

**Comment by koshkarov** [comment]
Due to constant issues with the tool and inability to actually use it, we decided to abandon it and found a better alternative.

**Comment by thallgren** [comment]
Closed by #3489
